mmv1/third_party/terraform/functions/project_id_from_self_link_test.go.erb

```diff
@@ -0,0 +1,147 @@
+<% autogen_exception -%>
+package functions_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccProviderFunction_project_id_from_self_link(t *testing.T) {
+	t.Parallel()
+	acctest.SkipIfVcr(t) // Need to determine if compatible with VCR, as functions are implemented in PF provider
```

Comment:
While waiting for initial review, I'll look into this!

Reply:
I've run a test locally with VCR_MODE = RECORDING and REPLAYING ok (after disabling the skip) and it was ok. I believe this is because although the provider-defined functions are in the plugin-framework part of the provider the API interactions (i.e. resources used in the acc tests) are all implemented with the SDK still.

(When I do the same with `TestAccDataSourceDnsManagedZone_basic`, which is affected by https://github.com/hashicorp/terraform-provider-google/issues/14158 , I see the VCR recording file contains only the last interaction and not a list of all API interactions. This isn't the case for the provider-defined functions tests, where the user agent for all interactions indicates the SDK is in use. )